pkg/server/download.go

```diff
@@ -293,7 +293,13 @@ func (s *server) downloadFileFromPresignedUrl(ctx context.Context, url string, t
 		return 0, err
 	}
 
-	err = os.Rename(tempPath, targetLocation)
+	err = deviceSafeRename(tempPath, targetLocation)
+
+	// Catch the error, read in the data from tempPath and write to targetLocation
+	if err != nil {
+		log.Fatal("Moving file failed")
+		log.Fatal(err)
```

Comment:
I'm not sure both of these will run.. I thought `log.Fatal` does an `os.exit(1)`? you might want to just interpolate the error into the first log
```
log.Fatal("Moving file failed: %v", err)
```